variable/include/scipp/variable/transform.h

```diff
@@ -472,20 +472,19 @@ template <bool dry_run> struct in_place {
         detail::dispatch_inner_loop<true>(op, indices.get(), inner_strides,
                                           inner_size, std::forward<T>(arg),
                                           std::forward<Ts>(other)...);
-        indices.increment_inner_by(inner_size);
-        indices.increment_outer();
+        indices.increment_by(inner_size != 0 ? inner_size : 1);
       }
     };
     if (begin.has_stride_zero()) {
       // The output has a dimension with stride zero so parallelization must
       // be done differently. See parallelization in accumulate.h.
-      auto indices = begin; // copy so that run doesn't modify begin
-      auto end = begin;
+      auto indices = begin;
+      auto end = std::move(begin);
```

Comment:
Nothing can be moved here I suppose? Might even prevent compiler from optimizing this out?